src/linux/mod.rs

```diff
@@ -165,11 +182,46 @@ impl<'a> SharedLibraryTrait for SharedLibrary<'a> {
 
     #[inline]
     fn name(&self) -> &CStr {
-        self.name
+        &*self.name
     }
 
-    #[inline]
     fn id(&self) -> Option<SharedLibraryId> {
+        fn align(alignment: usize, offset: &mut usize) {
+            let diff = *offset % alignment;
+            if diff != 0 {
+                *offset += alignment - diff;
+            }
+        }
+
+        unsafe {
+            for segment in self.segments() {
+                if segment.phdr.as_ref().unwrap().p_type != libc::PT_NOTE {
+                    continue;
+                }
+
+                let mut offset = (*segment.phdr).p_offset as usize;
+                let mut alignment = (*segment.phdr).p_align as usize;
+                if alignment < 4 {
+                    alignment = 4;
+                }
+                let size = offset + (*segment.phdr).p_filesz as usize;
+
+                while offset < size {
+                    let nhdr: &Nhdr = mem::transmute(self.addr.offset(offset as isize));
```

Comment:
To fix clippy warning: `let nhdr = &*((self.addr as usize + offset) as *const Nhdr);`